core/src/main/java/com/google/cloud/sql/core/DefaultConnectionInfoRepository.java

```diff
@@ -298,7 +304,13 @@ private InstanceMetadata fetchMetadata(CloudSqlInstanceName instanceName, AuthTy
 
         logger.debug(String.format("[%s] METADATA DONE", instanceName));
 
-        return new InstanceMetadata(ipAddrs, instanceCaCertificate);
+        return new InstanceMetadata(
+            instanceName,
+            ipAddrs,
+            Collections.singletonList(instanceCaCertificate),
+            "GOOGLE_MANAGED_CAS_CA".equals(instanceMetadata.getServerCaMode()),
+            instanceMetadata.getDnsName(),
+            pscEnabled);
```

Comment:
did we decide to have this here? I thought we were going to use the presence of the PSC ip type in the map to check PSC enabled?